### PR TITLE
Migrate show method to netmiko base

### PR DESCRIPTION
**Pull request by jmcgill298**

### CODE REVIEW

**Review by jvanderaa — APPROVED** [review]
LGTM

File: pyntc/devices/base_netmiko.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-
-# from pyntc.errors import CommandError, CommandListError
+from ..errors import CommandError, CommandListError
 
 
 @fix_docs
 class BaseNetmikoDevice(BaseDevice):
     """Netmiko Base Driver."""
 
+    error_keywords = ["% ", "Error:", "Incorrect usage"]
```
Comment (jvanderaa) [inline pyntc/devices/base_netmiko.py:15]: Consider a set? Not really a requirement, but an idea here.

**Review by jvanderaa — APPROVED** [review]
LG, just the one comment that may not be necessary.

File: test/unit/conftest.py [hunk 2]
```diff
@@ -327,7 +335,83 @@ def _mock(side_effects, existing_device=None, device=ios_device):
...
+        with mock.patch.object(device, "_send_commands", autospec=True) as mock_send_commands:
+            mock_send_commands.side_effect = get_side_effects(netmiko_mock_path, side_effects)
+        device._send_commands = mock_send_commands
         return device
+
+    return _mock
+
+
+@pytest.fixture
+def netmiko_show(netmiko_device, netmiko_mock_path):
```
Comment (mundruid) [inline test/unit/conftest.py:377]: Are you testing for single item (strings) vs multi-item?

**Review by mundruid — COMMENTED** [review]
This looks really good, I am glad we are working towards this `base_netmiko`, it was a good idea.I had a few comments, mainly we may need to work on more robust tests but I may not completely understand what you are testing.

File: pyntc/devices/base_netmiko.py [hunk 3]
```diff
@@ -127,3 +194,82 @@ def open(self, confirm_active: bool = True, **netmiko_args: Any) -> None:
...
+            >>> device = IOSDevice(**connection_args)
+            >>> version = device.show("show version")
+            >>> print(version)
+            'Cisco IOS Software, ...'
+            >>> version = device.show(["show version", "show inventory"])
+            >>> print(version)
+            ['Cisco IOS Software, ...', 'NAME: "chassis", DESCR: ...']
+            >>>
+        """
+        original_command_is_str: bool = isinstance(command, str)
```
[the remark this answers is not in the document]
Comment (jmcgill298) [reply pyntc/devices/base_netmiko.py:229]: 1) The current api for `show` returns a string, and this maintains backwards compatibility
2) As a user, I would expect that passing a string in would result in having a string returned to me (maybe that is just me, but that is part of my answer as to why I am doing this)
3) A user can have consistent return values by always passing a list to the method

File: pyntc/devices/base_netmiko.py [hunk 4]
```diff
@@ -127,3 +194,82 @@ def open(self, confirm_active: bool = True, **netmiko_args: Any) -> None:
...
+
+        if original_command_is_str:
+            command = [command]  # type: ignore [list-item]
+
+        try:
+            command_responses = self._send_commands(self.native.send_command, command, netmiko_args)
+        # TODO: Remove this when deprecating CommandListError
+        except CommandError as err:
+            if not original_command_is_str:
```
[the remark this answers is not in the document]
Comment (jmcgill298) [reply pyntc/devices/base_netmiko.py:238]: This is short lived until *_list methods are removed, and then we will always raise `CommandError`. This PR also updates the `CommandError` Exception Class in preparation for this. As it is, we can just delete these lines once we remove *_list.

File: test/unit/conftest.py (hunk 2 above)
Comment (jmcgill298) [reply test/unit/conftest.py:377]: This only supports passing a single command currently. I started to update `get_side_effects` to support multiple commands, but then I decided that wasn't a current requirement, and that I should hold off on increasing the complexity until it was needed.

**Comment by jmcgill298** [comment]
Thanks for the review @mundruid I have responded to the questions raised. I am curious as what you believe is missing from the unit tests?

File: pyntc/devices/base_netmiko.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mundruid) [reply pyntc/devices/base_netmiko.py:238]: Makes sense. Thanks @jmcgill298

**Review by mundruid — APPROVED** [review]
I missed the `test_base_netmiko.py` when I was doing my review @jmcgill298 . Thanks for answering my questions. I think it looks good.

**Comment by jmcgill298** [comment]
thanks @jvanderaa @mundruid

**Pull request merged**